### ISSUE TITLE
`indent`: regression for class property initialization between 2.1.0 and 2.2.0

### ISSUE THREAD

**Issue by orgads** [eslint-stylistic/eslint-stylistic#429]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

```ts
class Foo {
  foo =
    'bar'; // Expected indentation of 2 spaces but found 4.
};
```

This worked correctly in 2.1.0, and broke in 2.2.0. Bisect leads to #416.

### Reproduction

https://github.com/eslint-stylistic/eslint-stylistic/pull/428

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [X] I am willing to submit a PR with failing tests
